napari/_qt/qt_viewer.py

```diff
@@ -523,6 +526,19 @@ def console(self, console):
             self.dockConsole.setWidget(console)
             console.setParent(self.dockConsole)
 
+    @ensure_main_thread
+    def _on_slice_ready(self, event):
+        responses = event.value
+        for layer, response in responses.items():
+            # Update the layer slice state to temporarily support behavior
+            # that depends on it.
+            layer._update_slice_response(response)
+            # The rest of `Layer.refresh` after `set_view_slice`, where `set_data`
+            # notifies the corresponding vispy layer of the new slice.
+            layer.events.set_data()
+            layer._update_thumbnail()
+            layer._set_highlight(force=True)
```

Comment:
So am I understanding correctly tha tthis whole method won't be needed once layers lose all the slice-related info?

Reply:
We'll still need a method like this since we need to do something with the responses. In [the NAP](https://napari.org/dev/naps/4-async-slicing.html#hooking-up-the-viewer), we'd replace `layer._update_slice_response` and `layer.events.set_data` with `layer_to_visual[layer].update_slice_response`. I think that's a good long term goal, but I avoided introducing that here to avoid temporarily complicating the logic.

We also need to find a way to deal with `update_thumbnail` (likely by creating the thumbnail in the slicing thread, then setting it here for the qt widget), and `_set_highlight` (maybe handle it from the vispy layer?).